### PR TITLE
GATEWAY-3345: my own number

### PR DESCRIPTION
**Pull request by juliameng77**
GATEWAY-3345: my own number

### CODE REVIEW

**Comment by milldaulagala** [comment]
APPROVED

**Review by david-porter-messagemedia — APPROVED** [review]

File: spec/openapi.yaml [hunk 1]
```diff
@@ -1433,11 +1441,85 @@ paths:
...
+        3. A unique verification code will be sent to the number
+        4. Submit an approval request containing the verification code form that number. A 200 OK response will indicate the number has been verified and is ready for use.
+
+        ⚠️ Own numbers need to be re-verified every 12 months. You will be notified by email that verification of your number is about to expire.
+        
+        ## The most basic request has the following structure:
           - ```sender_address``` is case insensitive. When one already exists for this account, you get a error message ```conflict```;
           - ```destination_countries``` currently we only support ```AU``` and ```GB``` as the destination country;
-          - ```sender_address_type``` and ```usage_type``` must be ```ALPHANUMERIC```;
+          - ```sender_address_type``` and ```usage_type``` must be ```ALPHANUMERIC```and ```ALPHANUMERIC``` for alpha tag, ```INTERNATIONAL```and ```OWN_NUMBER``` for my own number;
```
Comment (paultirta) [inline spec/openapi.yaml]: I don't really understand this sentence. Can we make it a bit clearer?

File: spec/openapi.yaml [hunk 2]
```diff
@@ -1433,11 +1441,85 @@ paths:
         - Source Address
       summary: Request a sender address
       description: |
-        The most basic request has the following structure:
+        ## Sender address is alpha tag
```
Comment (arcdyc-sinch) [inline spec/openapi.yaml]: Sender address is _an_ alpha tag

File: spec/openapi.yaml [hunk 3]
```diff
@@ -1433,11 +1441,85 @@ paths:
...
+        * 6-11 characters long
+        * Only contains characters from the following sets:
+        * A-Z
+        * a-z
+        * 0-9
+        * _ (underscore)
+        * \- (hyphen)
+        
+        ## Sender address is my own number
+        Before you can send messages using a your own number (i.e. a non-dedicated or rotary number), you need to verify that you have a right to use that number. Ensuring you have a right to use a phone number is an important ACMA regulatory requirement, aiming to prevent scam, spam and misuse of messaging services.
```
Comment (arcdyc-sinch) [inline spec/openapi.yaml]: (i.e. a non-dedicated _nor a_ rotary number)

File: spec/openapi.yaml [hunk 4]
```diff
@@ -1433,11 +1441,85 @@ paths:
...
+        
+        If you do not wish to verify a number from within the web portal, or need to verify a large quantity of numbers, you can verify a number programmatically - via API.
+        
+        ### The process to verify a number via the API is:
+        1. (If you don’t have existing credentials) - Create Basic or HMAC credentials for the API by going to https://hub.messagemedia.com/api-settings
+        2. Submit a verification request with the number and destination country (usually AU) as per below. **Note**: *the verification is for the account associated with the API credentials only.*
+        3. A unique verification code will be sent to the number
+        4. Submit an approval request containing the verification code form that number. A 200 OK response will indicate the number has been verified and is ready for use.
```
Comment (arcdyc-sinch) [inline spec/openapi.yaml]: _for_ that number.

File: spec/openapi.yaml [hunk 5]
```diff
@@ -1433,11 +1441,85 @@ paths:
...
+
+        ⚠️ Own numbers need to be re-verified every 12 months. You will be notified by email that verification of your number is about to expire.
+        
+        ## The most basic request has the following structure:
           - ```sender_address``` is case insensitive. When one already exists for this account, you get a error message ```conflict```;
           - ```destination_countries``` currently we only support ```AU``` and ```GB``` as the destination country;
-          - ```sender_address_type``` and ```usage_type``` must be ```ALPHANUMERIC```;
+          - ```sender_address_type``` and ```usage_type``` must be ```ALPHANUMERIC```and ```ALPHANUMERIC``` for alpha tag, ```INTERNATIONAL```and ```OWN_NUMBER``` for my own number;
           - ```reason``` for requesting a sender address is required;
+            - when requesting my own number: provide your reason to use this number in a sentence.e.g. “This is my personal number”.
```
Comment (arcdyc-sinch) [inline spec/openapi.yaml]: sentence. e.g.

(space between _sentence._ and _e.g._)

File: spec/openapi.yaml [hunk 6]
```diff
@@ -1433,11 +1441,86 @@ paths:
...
+              - `COMPANY_NAME`
+              - `PARTNERSHIP_NAME`
+              - `REGISTERED_TRUST_NAME`
+              - `CO_OPERATIVE_NAME`
+              - `INDIGENOUS_CORPORATION_NAME`
+              - `REGISTERED_ORGANISATION_NAME`
+              - `PERSONAL_NAME`
+              - `AUSTRALIAN_TRADEMARK`
+              - `INTERNATIONAL_TRADEMARK`
+              - `ASUTRALIAN_GOVERNMENT_AGENCY_OR_ENTITY`
```
Comment (arcdyc-sinch) [inline spec/openapi.yaml]: AUSTRALIAN_GOVERNMENT_AGENCY_OR_ENTITY

File: spec/openapi.yaml [hunk 7]
```diff
@@ -1433,11 +1441,86 @@ paths:
...
+        3. A unique verification code will be sent to the number
+        4. Submit an approval request containing the verification code form that number. A 200 OK response will indicate the number has been verified and is ready for use.
+
+        ⚠️ Own numbers need to be re-verified every 12 months. You will be notified by email that verification of your number is about to expire.
+        
+        ## The most basic request has the following structure:
           - ```sender_address``` is case insensitive. When one already exists for this account, you get a error message ```conflict```;
           - ```destination_countries``` currently we only support ```AU``` and ```GB``` as the destination country;
-          - ```sender_address_type``` and ```usage_type``` must be ```ALPHANUMERIC```;
+          - Ensure that the `sender_address_type` parameter is set to `ALPHANUMERIC` for alpha tags and `INTERNATIONAL` for my own number;
```
Comment (arcdyc-sinch) [inline spec/openapi.yaml]: Can be made as a table

```
| Description   | `sender_address_type` | `usage_type`   |
|---            |---                    |---             |
| Alpha tag     | `ALPHANUMERIC`        | `ALPHANUMERIC` |
| My own number | `INTERNATIONAL`       | `OWN_NUMBER`   |
```

File: spec/openapi.yaml [hunk 8]
```diff
@@ -1433,11 +1441,86 @@ paths:
...
+              - `CONTRACTION_OF_NAME`
+              - `OTHER`
+
+            - `description`: Description for selecting OTHER as the use case. Limited to 200 characters.
+
+            - `email`: The preferred contact email for our approval team when additional details are required.
+
+            - `australianGovernmentAgencyOrEntityName`: The name of the entity.
+
+            - `abn`: Australia business number
```
Comment (arcdyc-sinch) [inline spec/openapi.yaml]: Australian business number

File: spec/openapi.yaml (hunk 1 above)
Comment (juliameng77) [reply spec/openapi.yaml]: Thanks, fixed

File: spec/openapi.yaml (hunk 8 above)
Comment (juliameng77) [reply spec/openapi.yaml]: Thanks, all fixed

**Review by arcdyc-sinch — APPROVED** [review]

File: spec/openapi.yaml (hunk 8 above)
Comment (milldaulagala) [reply spec/openapi.yaml]: APPROVED

File: spec/openapi.yaml [hunk 9]
```diff
@@ -1433,11 +1441,90 @@ paths:
...
+        3. A unique verification code will be sent to the number
+        4. Submit an approval request containing the verification code for that number. A 200 OK response will indicate the number has been verified and is ready for use.
+
+        ⚠️ Own numbers need to be re-verified every 12 months. You will be notified by email that verification of your number is about to expire.
+        
+        ## The most basic request has the following structure:
           - ```sender_address``` is case insensitive. When one already exists for this account, you get a error message ```conflict```;
           - ```destination_countries``` currently we only support ```AU``` and ```GB``` as the destination country;
-          - ```sender_address_type``` and ```usage_type``` must be ```ALPHANUMERIC```;
+          - Ensure that the `sender_address_type` parameter is set to `ALPHANUMERIC` for alpha tags and `INTERNATIONAL` for my own number;
```
Comment (arcdyc-sinch) [inline spec/openapi.yaml]: Please replace lines 1485 and 1486 to
```
`sender_address_type` and `usage_type` parameters are set to the following value pairs:
```

**Pull request merged**